client/src/main/java/com/flipkart/flux/client/intercept/WorkflowInterceptor.java

```diff
@@ -50,16 +52,18 @@
     @Inject
     private Provider<FluxRuntimeConnector> connectorProvider;
 
-    private final ObjectMapper objectMapper;
+    @Inject
+    private Provider<ObjectMapper> objectMapper;
 
     public WorkflowInterceptor() {
-        this.objectMapper = new ObjectMapper();
     }
 
-    public WorkflowInterceptor(LocalContext localContext, Provider<FluxRuntimeConnector> connectorProvider) {
+    @VisibleForTesting
```

Comment:
this is used to specify that a public method is only publicly available for testing. Should not be used in prod flows.